src/py/flwr/cli/build.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower command line interface `build` command."""
+
+import hashlib
+import os
+import zipfile
+from pathlib import Path
+from typing import Optional
+
+import jwt
+import pathspec
+import typer
+from typing_extensions import Annotated
+
+from .config_utils import validate_project_dir
+from .utils import is_valid_project_name
+
+
+# pylint: disable=too-many-locals
+def build(
+    directory: Annotated[
+        Optional[Path],
+        typer.Option(help="The Flower project directory to bundle into a FAB"),
+    ] = None,
+    signed: Annotated[bool, typer.Option(help="Flag to sign the FAB")] = False,
+) -> None:
+    """Build a Flower project into a FAB file.
+
+    It can be ran without any argument:
+
+        `flwr build`
+
+    In which case, it will try to bundle the current directory if it is a
+    valid Flower App.
+
+    It can also be ran with a directory argument:
+
+        `flwr build --directory ./docs/target_dir`
+
+    And finally, the `--signed` argument can also be used to sign the FAB:
+
+        `flwr build --signed`
+
+    Note that this will prompt the user for its private key.
```

Comment:
```suggestion
    """Build a Flower project into a Flower App Bundle (FAB).

    You can run `flwr build` without any arguments to bundle the current directory:

        `flwr build`

    You can also build a specific directory:

        `flwr build --directory ./projects/flower-hello-world`

    And finally, to sign the resulting `.fab` file, use the `--signed` argument:

        `flwr build --sign`

    Note that this will prompt you for its private key to use for signing.
```